File: web-shop-frontend/model/entity/auth/AuthenticationResponse.cs
using Newtonsoft.Json;
using web_shop_frontend.model.enums;

namespace web_shop_frontend.model.entity;

public class AuthenticationResponse
{
    [JsonProperty("accessToken")]
    public string AccessToken { get; set; }
    
    [JsonProperty("refreshToken")]
    public string RefreshToken { get; set; }
    
    [JsonProperty("username")]
    public string Username { get; set; }
    
    [JsonProperty("role")]
    public Role Role { get; set; }
    
    public AuthenticationResponse(string accessToken, string refreshToken, string username, Role role)
    {
        AccessToken = accessToken;
        RefreshToken = refreshToken;
        Username = username;
        Role = role;
    }
}

File: web-shop-backend/src/main/java/com/shadow_garden/webshopbackend/dto/auth/AuthenticationResponse.java
package com.shadow_garden.webshopbackend.dto.auth;

import com.shadow_garden.webshopbackend.enums.Role;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class AuthenticationResponse {
    private String accessToken;
    private String refreshToken;
    private String username;
    private Role role;
}

